src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -581,11 +581,11 @@ private void ProcessEvents()
                                     continue;
                                 }
                             }
-                            expandedName = associatedDirectoryEntry.GetPath(true, nextEvent.name);
+                            expandedName = associatedDirectoryEntry.GetPath(true, nextEvent.name).AsSpan();
```

Comment:
Nit: the `AsSpan()` shouldn't be necessary, due to the implicit cast from string to `ReadOnlySpan<char>`, but having it doesn't hurt.